src/Prototype/Database.hs

```diff
@@ -22,8 +25,10 @@ data Handle = Handle
     -- ^ Password, and User. Those are real users. We don't store the password
     -- in a specific data type to avoid manipulating it and risking sending it
     -- over the wire.
-  , hTodoLists :: TVar [(String, TodoList)]
-    -- ^ Associates namespaces to Todo lists.
+  , hTodoLists :: TVar (Map String [TVar TodoList])
```

Comment:
Why are we nesting `TVar`'s? 
(this is more of a question than a comment).

Reply:
The idea was that what we store are `TVar Something`, and such values can be shared in multiple structures (if you update one, you want the changes to be reflected in other locations). So `hTodoLists` is really just one way of accessing the Todo list but we could store them in additional locations (e.g. in a "favorite" list).

Another reason, even if we don't want sharing the value in multiple places, is to reduce the contention if multiple threads want to change different Todo list, but maybe I'm wrong.

That being said, the suggested structure in the blog post annoucing `stm-containers` is `TVar (HashMap (TVar a))` : http://nikita-volkov.github.io/stm-containers/ so I guess this matches what I was doing.

I'll see to convert the code to use that library ; this seems like a nice suggestion.